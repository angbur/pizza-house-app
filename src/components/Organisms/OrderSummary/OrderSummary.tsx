import Button from 'components/Atoms/Button/Button';
import Typography from 'components/Atoms/Typography/Typography';
import { MouseEventHandler } from 'react';
import styled from 'styled-components';
import OrderForm from '../OrderForm/OrderForm';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: start;
  text-align: center;
  & > button {
    margin: 2rem 0;
  }
  & .title {
    margin-bottom: 1rem;
  }
  @media (max-width: 766px) {
    justify-content: center;
  }
`;

type OrderSummaryProps = {
  data: OrderForm;
  onClick: MouseEventHandler<HTMLButtonElement>;
};

const OrderSummary = ({ data, onClick }: OrderSummaryProps) => {
  return (
    <>
      <Container>
        <Typography variant={'card-title'} className={'title'} color={'primary'}>
          Address:
        </Typography>
        <Typography variant={'paragraph'} color={'light'}>
          {data.firstName.concat(' ', data.lastName)}
        </Typography>
        <Typography variant={'paragraph'} color={'light'}>
          {data.email}
        </Typography>
        <Typography variant={'paragraph'} color={'light'}>
          {data.phone}
        </Typography>
        <Typography variant={'paragraph'} color={'light'}>
          {data.street.concat(' ', data.houseNumber)}
        </Typography>
        <Typography variant={'paragraph'} color={'light'}>
          {data.postalCode.concat(' ', data.city)}
        </Typography>
        <Button variant={'secondary-light'} onClick={onClick}>
          Edit
        </Button>
      </Container>
    </>
  );
};

export default OrderSummary;
