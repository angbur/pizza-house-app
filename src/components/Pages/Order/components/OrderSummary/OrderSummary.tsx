const OrderSummary = () => {
  return null;
};

export default OrderSummary;
