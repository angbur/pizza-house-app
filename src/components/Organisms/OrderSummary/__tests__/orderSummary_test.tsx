import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import OrderSummary from '../OrderSummary';

const data = {
  firstName: 'Kamila',
  lastName: 'Lasocka',
  email: 'lasocka@gmail.com',
  city: 'WrocÅ‚aw',
  postalCode: '55-001',
  street: 'Malinowa',
  houseNumber: '3a',
  phone: '697748282',
};

const Wrapper = () => {
  return (
    <ThemeProvider>
      <OrderSummary data={data} onClick={jest.fn()} />
    </ThemeProvider>
  );
};

describe('Order Summary', () => {
  it('should renders order summary correctly', () => {
    const tree = renderer.create(<Wrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
