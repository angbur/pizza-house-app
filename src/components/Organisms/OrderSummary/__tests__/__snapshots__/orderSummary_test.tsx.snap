// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Summary should renders order summary correctly 1`] = `
<div
  className="sc-jrcTuL iJTmuF"
>
  <h3
    className="card-title title"
    style={
      Object {
        "color": "#EB853E",
        "fontFamily": "Klepon Ijo",
        "fontSize": "2rem",
        "fontWeight": "400",
        "letterSpacing": "2px",
        "lineHeight": "1.2",
      }
    }
  >
    Address:
  </h3>
  <p
    className="paragraph"
    style={
      Object {
        "color": "#FCFBFB",
        "fontFamily": "Source Sans Pro",
        "fontSize": "1rem",
        "fontWeight": "400",
        "letterSpacing": "0",
        "lineHeight": "1.6",
      }
    }
  >
    Kamila Lasocka
  </p>
  <p
    className="paragraph"
    style={
      Object {
        "color": "#FCFBFB",
        "fontFamily": "Source Sans Pro",
        "fontSize": "1rem",
        "fontWeight": "400",
        "letterSpacing": "0",
        "lineHeight": "1.6",
      }
    }
  >
    lasocka@gmail.com
  </p>
  <p
    className="paragraph"
    style={
      Object {
        "color": "#FCFBFB",
        "fontFamily": "Source Sans Pro",
        "fontSize": "1rem",
        "fontWeight": "400",
        "letterSpacing": "0",
        "lineHeight": "1.6",
      }
    }
  >
    697748282
  </p>
  <p
    className="paragraph"
    style={
      Object {
        "color": "#FCFBFB",
        "fontFamily": "Source Sans Pro",
        "fontSize": "1rem",
        "fontWeight": "400",
        "letterSpacing": "0",
        "lineHeight": "1.6",
      }
    }
  >
    Malinowa 3a
  </p>
  <p
    className="paragraph"
    style={
      Object {
        "color": "#FCFBFB",
        "fontFamily": "Source Sans Pro",
        "fontSize": "1rem",
        "fontWeight": "400",
        "letterSpacing": "0",
        "lineHeight": "1.6",
      }
    }
  >
    55-001 Wroc≈Çaw
  </p>
  <button
    className="sc-bcXHqe sc-hLBbgP feNVH joABVS secondary-light btn-lg button"
    onClick={[MockFunction]}
    style={
      Object {
        "fontSize": "1.05rem",
        "padding": "0.65rem 1.25rem",
      }
    }
  >
    Edit
  </button>
</div>
`;
